__tests__/app.test.js

```diff
@@ -347,3 +327,139 @@ describe("PATCH /api/reviews/:review_id", () => {
   });
 });
+
+describe("GET /api/users", () => {
+  test("status 200, returns all users", () => {
+    return request(app)
+      .get("/api/users")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.users.length).toBe(4);
+        body.users.forEach((user) => {
+          expect(user).toMatchObject({
+            username: expect.any(String),
+            name: expect.any(String),
+            avatar_url: expect.any(String),
+          });
+        });
+      });
+  });
+});
+
+describe("GET /api/reviews (queries)", () => {
+  test("status 200, if given a category query only returns result with that category and order by date", () => {
+    return request(app)
+      .get("/api/reviews?category=dexterity")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.reviews.length).toBe(1);
+        expect(body.reviews).toBeSortedBy("created_at", {
+          descending: true,
+        });
+        expect(body.reviews[0]).toMatchObject({
+          review_id: 2,
+          title: "Jenga",
+          category: "dexterity",
+          designer: "Leslie Scott",
+          owner: "philippaclaire9",
+          review_body: "Fiddly fun for all the family",
+          review_img_url:
+            "https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png",
+          created_at: expect.any(String),
+          votes: 5,
+          comment_count: 3,
+        });
+      });
+  });
+  test("status 200, if given no category returns all results", () => {
+    return request(app)
+      .get("/api/reviews")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.reviews.length).toBe(13);
+      });
+  });
+  test("status 404, if category doesn't exist", () => {
+    return request(app)
+      .get("/api/reviews?category=dexterity=banana")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("not found");
+      });
+  });
+  test("status 200, if category exists but not in review table", () => {
+    return request(app)
+      .get("/api/reviews?category=children's games")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.reviews.length).toBe(0);
+      });
+  });
+  test("status 200, if an order ASC by query is given sorts the list", () => {
+    return request(app)
+      .get("/api/reviews?order=ASC")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.reviews.length).toBe(13);
+        expect(body.reviews).toBeSortedBy("created_at", {
+          descending: false,
+        });
+      });
+  });
+  test("status 200, if a order DESC by query is given sorts the list", () => {
+    return request(app)
+      .get("/api/reviews?order=DESC")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.reviews.length).toBe(13);
+        expect(body.reviews).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
+  });
+  test("status 200, if invalid order parameter given defaults to DESC", () => {
```

Comment:
Hi David, really good testing! On this one however, for an invalid order query it should be a 400 👍